src/main/java/dev/chuckbot/controller/JokeController.java

```diff
@@ -32,23 +35,36 @@ public JokeController(ChuckNorrisJokeService service) {
     public String displayIndex(Model model, @RequestParam(required = false, defaultValue = "") String sort) {
         List<Joke> jokeList = service.getAllJokes();
 
+        // Sort the jokelist ascendind or descending by creation date or joketext
         if (sort.equals("date")) {
-            Collections.sort(jokeList, new CreationDateComparator().reversed());
+            if (sortCount % 2 == 0) {
+                jokeList.sort((Comparator.comparing(Joke::getCreationDate).reversed()));
+            } else {
+                jokeList.sort((Comparator.comparing(Joke::getCreationDate)));
+            }
             model.addAttribute("jokes", jokeList);
+            sortCount++;
         } else if (sort.equals("alphabetically")) {
-            Collections.sort(jokeList, new JokeTextComparator());
+            if (sortCount % 2 == 0) {
```

Comment:
This cannot work for different users as the "state" of the sort is thread-specific, see diagram.
![image](https://user-images.githubusercontent.com/2828581/142585284-05f094cf-7db2-4565-8a59-f21f71e761bd.png)